sdk/releases.md

```diff
@@ -0,0 +1,36 @@
+# Ably SDK Team: Guidance on Releases
+
+This document outlines our ideal requirements for SDK releases.
+
+There will be some SDK repositories:
+
+- which are yet to have the work done on them which will implement this
+- where it's necessary to deviate from some of the ideals specified here due to constraints which are innate or otherwise idiomatic to the platform being released for
+
+Where technically possible, SDK repository release processes which claim to adhere to this guidance must follow these ideal requirements closely.
+
+## Release Process
+
+1. Merge all pull requests containing changes intended for this release to `main` branch
+2. Prepare a [Release Branch](#release-branch) and a corresponding pull request, obtain approval from reviewers and then merge to `main` branch
+3. Trigger the [Publish Workflow](#publish-workflow)
+4. If the publish fails then add commits to the `main` branch that aim to fix the problem and then try step 3 again
+5. Push Git version tag, which may or may not have a `v` prefix according to the existing conventions for this SDK repository
```

Comment:
Does the tag & release not need to come before the publish workflow? I'm not familiar with the details of how publishing to a package repository works, but I imagine that in some cases we need to refer to an existing tag when we publish?

Reply:
I recognise that there will be instances, hence [this statement](https://github.com/ably/engineering/blob/e712b93b53342ad51b493391c26e4e5ec1ec9719/sdk/releases.md?plain=1#L8) earlier on in this document:

> where it's necessary to deviate from some of the ideals specified here due to constraints which are innate or otherwise idiomatic to the platform being released for

However, generally, our preference should be to successfully make the release _before_ pushing the tag. This avoids any requirement for us to do messy things like moving tags.

Reply:
OK, cool 👍